app/models/agent.rb

```diff
@@ -44,7 +44,7 @@ class Agent < ActiveRecord::Base
   after_save :possibly_update_event_expirations
 
   belongs_to :user, :inverse_of => :agents
-  belongs_to :service, :inverse_of => :agents
+  belongs_to :service, :inverse_of => :agents, optional: true
```

Comment:
It could be, but when DJ is active it does not matter and can not be tested (https://github.com/collectiveidea/delayed_job_active_record/issues/128).